awsbestpractices/missing-pagination/MissingPagination.java

```diff
@@ -0,0 +1,507 @@
+            listObjectsV2Result = s3Client.listObjectsV2(listObjectsV2Request);
+        }
+
+
+        return listObjectsV2Result.getObjectSummaries().stream()
+                .map(S3ObjectSummary::getKey)
+                .collect(Collectors.toList());
+    }
+
+    // Missing
+    // Adapted from
+    // https://code.amazon.com/packages/BrowseItemClassificationServiceLambdaJava/blobs/5eaf5c68ba4188fe786074cc1921f2af04d00be5/--/src/com/amazon/catalog/browse/bics/lambda/handler/MetricsQueryResultHandler.java#L41
+    public Set<String> missingPagination_4(Event event, Context context) throws InvalidRequestException {
+
+        checkNotNull(event, "Event cannot be null.");
+        checkNotNull(event.getExecutionContext(), "Event execution context cannot be null.");
+
+        final String executionContext = event.getExecutionContext();
+        if (executionContext.equals(EXECUTION_CONTEXT_CAC_ONLY)
+                || executionContext.equals(EXECUTION_CONTEXT_BOTH)) {
+
+            if (event.getPluginConfig() != null && event.getPluginConfig().containsKey(QUERY_EXECUTION_ID)) {
+
+                // Get Athena client
+                AmazonAthena client = null;
+                try {
+                    client = AthenaClientProvider.getAmazonAthena();
+                } catch (Exception e) {
+                    context.getLogger().log("Unexpected exception" + e);
+                    throw new IllegalStateException(e);
+                }
+
+                // Get the query result set
+                GetQueryResultsRequest queryResultsRequest = new GetQueryResultsRequest().withQueryExecutionId(event.getPluginConfig().get
+                        (QUERY_EXECUTION_ID));
+                GetQueryResultsResult queryResultsResult = client.getQueryResults(queryResultsRequest);
+                ResultSet resultSet = queryResultsResult.getResultSet();
+
+                // Extract the required metrics
+                // Update the metric metadata and data points
+                event.setGenerationDate(DateTimeUtils.getDateTimeNow());
+                EventUtils.addDataPoints(QueryManager.getMetricsResult(resultSet), event, CATC_PROGRAM_NAME, CATC_RAW_DATASET);
+            }
+
+            event.setGenerationDate(DateTimeUtils.getDateTimeNow());
+            return event;
+        }
+
+        return event;
+    }
+
+    // Missing
+    // Adapted from
+    // https://code.amazon.com/packages/WellArchitectedExternalService/blobs/ea925ff83263e1dca79cfe9495a3b13e269cd725/--/src/com/amazon/wellarchitectedexternalservice/lambda/util/S3ClientUtils.java#L23
+    // Tricky because they check isEmpty() and size(), but pagination is still needed.
+    public void missingPagination_5(final AmazonS3 s3Client, final String bucketName,
+                                    final String pattern) {
+        final List<String> objectList = s3Client.listObjectsV2(new ListObjectsV2Request()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.